librus/__init__.py

```diff
@@ -253,12 +254,17 @@ def __init__(self, day, index, subject, time, teacher, classroom):
 
 
 class Message(object):
-    def __init__(self, sender, subject, sent_at):
+    def __init__(self, message_id, sender, subject, sent_at, is_read):
+        self.message_id = message_id
         self.sender = sender
         self.subject = subject
         self.sent_at = sent_at
+        self.is_read = is_read
         self.content = None
 
+    def __eq__(self, other):
+        return self.message_id == other.message_id
```

Comment:
This operator is useful when you add messages to the set. Messages with the same id are the same messages, so they only appear once in the set.

Reply:
There shouldn't be duplicates, but if you want to do deduplication you can always collect them into a dict, using message_id as the key.
i am concerned that someone will lose their messages silently if Librus's urls change and we cannot extract message_id anymore.

Reply:
OK, what if `message_id` was simply a url? It unambiguously identifies the message, so it's apparently a good indicator of the equality of two messages.

Reply:
works for me

Reply:
I mis-read the conversation. I understood the proposal to treat URL as the message id is to avoid parsing the url (with`re`, split or offset based), and not in order to keep __eq__. Sorry for miscommunication from my side.